subsys/logging/Kconfig

```diff
@@ -424,6 +424,52 @@ config LOG_BACKEND_RTT_FORCE_PRINTK
 
 endif # LOG_BACKEND_RTT
 
+config LOG_BACKEND_SPINEL
+	bool "Enable OpenThread dedicated Spinel protocol backend"
+	depends on (OPENTHREAD_NCP_SPINEL_ON_UART_DEV_NAME!=UART_CONSOLE_ON_DEV_NAME || !LOG_BACKEND_UART)
+	depends on NET_L2_OPENTHREAD
+	help
+	  When enabled, backend will use OpenThread dedicated SPINEL protocol for logging.
+	  This protocol is byte oriented and wrapps given messages into serial frames.Backend
+	  should be enabled only to OpenThread purposes and when UART backend is disabled
+	  or works on antoher UART device to avoid interference.
+
+if LOG_BACKEND_SPINEL
+
+config LOG_BACKEND_SPINEL_BUFFER_SIZE
+	int "Size of reserved up-buffer for logger output."
+	default 1024
```

Comment:
it doesn't seem to make sense to have this default that high. Data from this buffer is passed later on to the ring buffer. I think that you won't see much performance difference if it is reduced to 16.

Reply:
Ok, I lowered value to 64, because when set to 16 there were many errors saying that operation has been prevented due to memory pressure. I think 16 bytes buffer is to small to send longer log messages, which are cut into many smaller pieces then.

Reply:
that rather shows that system is not ready for pressure and imo this should be investigated further. When buffer is smaller then flushing function is called multiple times for single log message, as you mentioned. Increasing it to 64 will lead to the situation where longer messages will be affected but problem may still exist.

I will give +1 for this one but i guess there will be improvements coming.